File: sources/icloud-workspace/seis-digital-experience-foundation/UIX-Apps-codex-delivery/polyglot/cpp/readiness_signal.cpp
#include <string>

struct ReadinessSignal {
  int languageCount;
  bool remoteBlocked;

  bool ok() const {
    return languageCount >= 12;
  }

  std::string status() const {
    if (remoteBlocked) {
      return "ready-with-blocker";
    }
    return ok() ? "ready" : "needs-work";
  }
};

File: sources/icloud-workspace/seis-digital-experience-foundation/runtime/server-upload-bundle/source/polyglot/csharp/ReadinessSignal.cs
namespace UIXApps.Polyglot;

public sealed record ReadinessSignal(int LanguageCount, bool RemoteBlocked)
{
    public bool Ok => LanguageCount >= 12;

    public string Status => RemoteBlocked
        ? "ready-with-blocker"
        : Ok ? "ready" : "needs-work";
}
